akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala

```diff
@@ -97,6 +98,19 @@ import akka.util.JavaDurationConverters._
     }(responseTimeout, ClassTag[Res](resClass))
   }
 
+  // Scala API impl
+  def pipeToSelf[Value](future: Future[Value], mapValue: Try[Value] ⇒ T): Unit = {
+    future.onComplete(self ! mapValue(_))
```

Comment:
this should be implemented in the same way as `ask`, so that the `mapValue` function is run by the actor, for additional thread safety

Reply:
I hope that just means adding `(system.executionContext)`, because that's what I understood.

Reply:
no, see how context.ask is implemented, it sends a message with the function to the actor, and the function is applied when that message is received (internal message handling).

Reply:
Got it.  Thanks.